src/util/operations.tsx

```diff
@@ -13,6 +13,26 @@ export const getInstanceName = (operation?: LxdOperation): string => {
   );
 };
 
+export const getSnapshotName = (operation?: LxdOperation): string => {
+  // the url can be one of below formats
+  // instance and volume snapshot operations are mutually exclusive
+  // /1.0/instances/<instance_name>/snapshots/<snapshot_name>
+  // /1.0/storage-pools/<pool_name>/volumes/custom/<volume_name>/snapshots/<snapshot_name>
+  const instanceSnapshots = operation?.resources?.instances_snapshots ?? [];
+  const storageVolumeSnapshots =
+    operation?.resources?.storage_volume_snapshots ?? [];
+
+  if (instanceSnapshots.length) {
+    return instanceSnapshots[0].split("/")[5].split("?")[0];
+  }
+
+  if (storageVolumeSnapshots.length) {
+    return storageVolumeSnapshots[0].split("/")[8].split("?")[0];
+  }
+
+  return "";
+};
+
```

Comment:
please add a spec test for this one, similar to the other functions in this file.

Reply:
Also, it might be better to split this by instance and volume snapshots. Because the logic has no relevant overlap and the use cases are distinct.

Reply:
sounds good, will do both. Let's stay consistent with the pattern to have separate functions/components for instance and volume snapshots